### ISSUE TITLE
Inquiry About Training Strategies

### ISSUE THREAD

**Issue by cjc2412** [XingfuWang/CSNet#3]
Hello,  I am writing to ask about the training strategies you employed in your experiments.
Specifically, I would like to know how you approached the selection of the final model and the reporting of results. Was the reported result the best accuracy achieved on the test set during training, or the accuracy after the final training epoch?
Understanding your approach would be very helpful for my own research, and I truly appreciate the effort you've put into making your work publicly available.

**Comment by cjc2412** [comment]
Because I noticed that your `main.py` script saves the highest score on the test set for each epoch, but I'm not entirely sure if that's the final reporting metric. I hope to confirm this with you.

**Comment by XingfuWang** [comment]
Hello, we report all experimental results in this paper consistently with the implementation in our source code. Thank you.

**Comment by cjc2412** [comment]
I noticed that the preprocessing in the provided example code includes a filtering step, but the paper does not mention filtering. I was wondering if, for the BCI data, the approach is the same as in the example code—specifically, reporting the highest score achieved during training epochs on the test set? Thank you!

**Comment by XingfuWang** [comment]
As you mentioned, the code provides a band-pass filtering option. However, in our experiments, we did not use band-pass filtering in order to preserve an end-to-end decoding pipeline. The filtering code is included only to offer readers an optional preprocessing choice. Empirically, CSNet achieves higher accuracy when band-pass filtering is not applied. In addition, the accuracy reported in the paper is consistent with the results produced by the code. Thank you!

**Comment by cjc2412** [comment]
Thanks for the clarification! Just to confirm: is the accuracy reported in the paper the best test accuracy over training epochs, or the test accuracy at the final epoch?
best_acc = 0 

for epoch in range(EPOCH):
    # ......

    model.eval()
    with torch.no_grad():

    # ......
    if test_accuracy >= best_acc:
        y_pred_best = y_pred          
        y_true_best = y_true         
        best_acc = test_accuracy    
        print('\033[91mBest_acc updated to {:.3f}\033[0m'.format(best_acc))
        cm = confusion_matrix(y_true.numpy(), y_pred.numpy())
        plot_confusion_matrix(cm=cm, classes=['feet', 'left', 'right', 'tongue'],
                              save_path=f'{save_path}/Confusion_matrix{k_f}.jpg')
        plt.close()
        torch.save(model, f'{save_path}/best_acc_model_k{k_f}.pth')
    if int(test_accuracy) == 1:
        break
np.savetxt(f'{save_path}/k{k_f}_acc{best_acc:.4f}.txt', [best_acc])
